Fix two snircd compile warnings.

channel.c: In function ‘SetAutoChanModes’:
channel.c:1344:17: warning: pointer targets in assignment differ in signedness [-Wpointer-sign]
was missing unsigned

./ircd_parser.y: In function ‘yyparse’:
./ircd_parser.y:1199:9: warning: pointer targets in passing argument 3 of ‘ipmask_parse’ differ in signedness [-Wpointer-sign]
../include/ircd_string.h:34:12: note: expected ‘unsigned char *’ but argument is of type ‘char *’
was missing unsigned and not initialized at 0

diff -r bb5955d8f128 ircd/channel.c
--- a/ircd/channel.c
+++ b/ircd/channel.c
@@ -1310,7 +1310,7 @@
 
 int SetAutoChanModes(struct Channel *chptr)
 {
-  static int chan_flags[] = {
+  static unsigned int chan_flags[] = {
     MODE_PRIVATE,       'p',
     MODE_SECRET,        's',
     MODE_MODERATED,     'm',
diff -r bb5955d8f128 ircd/ircd_parser.y
--- a/ircd/ircd_parser.y
+++ b/ircd/ircd_parser.y
@@ -1183,7 +1183,7 @@
 spoofitems '}' ';'
 {
   struct irc_in_addr ip;
-  char bits;
+  unsigned char bits = 0;
   int valid = 0;
 
   if (spoof->username == NULL && spoof->realhost) {
